src/Symfony/Bundle/SecurityBundle/SecurityUserValueResolver.php

```diff
@@ -40,7 +40,7 @@ public function __construct(TokenStorageInterface $tokenStorage)
     public function supports(Request $request, ArgumentMetadata $argument)
     {
         // only security user implementations are supported
-        if (UserInterface::class !== $argument->getType()) {
+        if (!$argument->getType() || !$this->implementsCorrectInterface($argument->getType())) {
```

Comment:
Should be in `Symfony\Component\Security\Http\Controller\UserValueResolver` instead (see deprecation notice above)

Reply:
It's in both classes

Reply:
It's in both classes :)

Reply:
Deprecated classes are frozen until they are suppressed, it should not get new features.